performance/admin.py

```diff
@@ -12,17 +171,49 @@ class EQPRSFileUploadAdmin(admin.ModelAdmin):
         "success",
         "updated_at",
     )
-    readonly_fields = (
-        "user",
-        "num_imported",
-        "num_not_imported",
-        "success",
-        "import_report",
+    fieldsets = (
+        ("", {
+            "fields": (
+                'user', 'file', 'import_report', 'num_imported', 'num_not_imported'
+            )
+        }),
     )
-    exclude = ("task",)
+
+    class Media:
+        js = ("js/eqprs-file-upload.js",)
+
+    def get_readonly_fields(self, request, obj=None):
+        if obj:  # editing an existing object
+            return ('user', 'file',) + self.readonly_fields
+        return self.readonly_fields
+
+    def render_change_form(self, request, context, add=False, change=False, form_url='', obj=None):
+        response = super(EQPRSFileUploadAdmin, self).render_change_form(request, context, add, change, form_url, obj)
+        response.context_data['title'] = 'EQPRS file upload' if response.context_data[
+            'object_id'] else 'Add EQPRS file upload'
+        return response
+
+    def save_model(self, request, obj, form, change):
+        super().save_model(request, obj, form, change)
+        full_text = obj.file.read().decode('utf-8')
+        report_type_validated = validate_report_type(full_text, obj.id)
+        if report_type_validated:
+            f = StringIO(full_text)
+            reader = csv.reader(f, delimiter=',')
+            parsed_data = [tuple(row) for row in reader]
+            parsed_data = [x for x in parsed_data if x]
+            del parsed_data[0:2]  # delete first 3 rows
+
+            frictionless_validated = validate_frictionless(parsed_data, obj.id)
+            if frictionless_validated:
+                del parsed_data[0]  # delete header row
+                threading.Thread(target=save_imported_indicators, args=(parsed_data, obj.id)).start()
```

Comment:
please use https://django-q.readthedocs.io/en/latest/tasks.html instead of starting a thread - I see where the misunderstanding came in :)

Reply:
this is why there's a task field in the upload model

Reply:
done 👍